isthmus/src/main/java/io/substrait/isthmus/SubstraitToCalcite.java

```diff
@@ -0,0 +1,119 @@
+package io.substrait.isthmus;
+
+import io.substrait.function.SimpleExtension;
+import io.substrait.relation.AbstractRelVisitor;
+import io.substrait.relation.NamedScan;
+import io.substrait.relation.Rel;
+import io.substrait.type.NamedStruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.jdbc.LookupCalciteSchema;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.RelBuilder;
+
+/**
+ * Converts between Substrait {@link Rel}s and Calcite {@link RelNode}s.
+ *
+ * <p>Can be extended to customize the {@link RelBuilder} and {@link SubstraitRelNodeConverter} used
+ * in the conversion.
+ */
+public class SubstraitToCalcite {
```

Comment:
Thinking about this some more, though I'm pretty happy with this I don't know if I'm "happy as public API forever" kind of happy. Could make this protected and keep it internal only for now.

Reply:
I think if you're uncertain, it might be best to be conservative and leave it protected until someone needs it.

Reply:
This got merged before I circled back to hiding this.

Looking at https://substrait.io/spec/versioning/, because we're pre-1.0 I'm more comfortable with having it in. It is actually helpful to have as downstream user and it does make it easier for folks experiment with converting substrait plans directly to Calcite.